### ISSUE TITLE
Port untilConcludes to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5933]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5933|
|Type|enhancement|
|Created|2012-09-02 21:46:39Z|
|Branch|https://github.com/twisted/twisted/tree/untilconcludes-python3-5933|

[ticket:5932 `twisted.python.log` depends on this].

<details><summary>Searchable metadata</summary>

```
trac-id__5933 5933
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_untilconcludes_python3_5933 branches/untilconcludes-python3-5933
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1346622399000000 1346622399000000
changetime__1346693051000000 1346693051000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [35483]) Branching to 'untilconcludes-python3-5933'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [35484]) Move the implementation to the py3k module; make the minor necessary syntax tweak.  Also fix the `__all__` of util.py to include this method.

refs [#5933](https://github.com/twisted/twisted/issues/5933)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/untilconcludes-python3-5933)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set owner to @exarkun|
|-|-|

Looks good, please merge (unless I get to it first).

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set status to `closed`|
|-|-|

(In [35499]) Merge untilconcludes-python3-5933 

Author: exarkun
Review: itamar
Fixes: [#5933](https://github.com/twisted/twisted/issues/5933)

Port twisted.python.util.untilConcludes to Python 3.
